tree.c

```diff
@@ -870,6 +879,9 @@ static void album_add_track(struct album *album, struct tree_track *track)
 
 	rb_link_node(&track->tree_node, parent, new);
 	rb_insert_color(&track->tree_node, &album->track_root);
+
+	album->num_tracks++;
+	album_for_each_track(album, ti_set_albumtracks, &album->num_tracks, 0);
```

Comment:
Isn't this an O(N^2) algorithm?

Reply:
Yes, it would be. The alternative would be to make the field itself a pointer, but I'm not sure if it's worth it since I doubt we'll be having hundreds of tracks per album.

Another option would be to modify the RB tree to include a count of tree children, but I doubt I'm not sure it's worth doing that.

The last option would be to make formatting support getting fields from a pointer to another struct (the album's num_tracks field), but that should probably be done separately (in another commit or PR).

Reply:
Since I understand the code a bit better after this draft, I can try it.